SegFault in ApplyPunishment

Hey guys we're hitting another seg fault on branch `main` [v0.1.45](https://github.com/ADimoska/SOMASExtended/releases/tag/v0.1.45)

It occurs in `ApplyPunishment` and I think the warning may be relevant here `Warning:  Dead agent should not be being kicked`

Like many seg faults, it does not occur on every run.

```console
Running turn for team  79de4e7d-5827-4158-a3f0-44ce476a6ec3
---------------------
d28f979c-4e80-4ee2-91d2-83faefdefd0e decides to ROLL AGAIN, last score: 8
d28f979c-4e80-4ee2-91d2-83faefdefd0e decides to [STICK], last score: 9
d28f979c-4e80-4ee2-91d2-83faefdefd0e's turn score: 17, total score: 197
Warning:  Dead agent should not be being kicked: 4225c13b-1f11-4be6-81cb-c19e5773b78e
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x148 pc=0x5c3fb6]

goroutine 1 [running]:
github.com/ADimoska/SOMASExtended/server.(*EnvironmentServer).ApplyPunishment(0xc0000f2000?, 0xc00009a1e0, {0x42, 0x25, 0xc1, 0x3b, 0x1f, 0x11, 0x4b, 0xe6, ...})
	/home/runner/work/SOMASExtended/SOMASExtended/server/EnvironmentServer.go:1138 +0x56
github.com/ADimoska/SOMASExtended/server.(*EnvironmentServer).RunTurnDefault(0xc0000f2000, 0xc00009a1e0)
	/home/runner/work/SOMASExtended/SOMASExtended/server/EnvironmentServer.go:99 +0x7bc
github.com/ADimoska/SOMASExtended/server.(*EnvironmentServer).RunTurn(0xc0000f2000, 0x0?, 0x3a)
	/home/runner/work/SOMASExtended/SOMASExtended/server/EnvironmentServer.go:374 +0x32d
github.com/MattSScott/basePlatformSOMAS/v2/pkg/server.(*BaseServer[...]).Start(0x68d3c0)
	/home/runner/go/pkg/mod/github.com/!matt!s!scott/base!platform!s!o!m!a!s/v2@v2.1.0/pkg/server/baseServer.go:111 +0x99
github.com/ADimoska/SOMASExtended/server.(*EnvironmentServer).Start(...)
	/home/runner/work/SOMASExtended/SOMASExtended/server/EnvironmentServer.go:637
main.main()
	/home/runner/work/SOMASExtended/SOMASExtended/main.go:80 +0x839
Error: Process completed with exit code 2.
```

--- a/server/EnvironmentServer.go
+++ b/server/EnvironmentServer.go
@@ -88,6 +88,7 @@ func (cs *EnvironmentServer) RunTurnDefault(team *common.Team) {
 		auditResult := team.TeamAoA.GetContributionAuditResult(agentToAudit)
 
 		if auditResult {
+			cs.ApplyPunishment(team, agentToAudit)
 			if team.TeamAoAID == 2 {
 				if agentToAudit == team.TeamAoA.(*common.Team2AoA).GetLeader() {
 					cs.ElectNewLeader(team.TeamID)
@@ -96,7 +97,6 @@ func (cs *EnvironmentServer) RunTurnDefault(team *common.Team) {
 					cs.RemoveAgentFromTeam(agentToAudit)
 				}
 			}
-			cs.ApplyPunishment(team, agentToAudit)
 		}
 
 		for _, agentID := range team.Agents {
@@ -159,6 +159,8 @@ func (cs *EnvironmentServer) RunTurnDefault(team *common.Team) {
 		auditResult := team.TeamAoA.GetWithdrawalAuditResult(agentToAudit)
 
 		if auditResult {
+			cs.ApplyPunishment(team, agentToAudit)
+
 			if team.TeamAoAID == 2 {
 				if agentToAudit == team.TeamAoA.(*common.Team2AoA).GetLeader() {
 					cs.ElectNewLeader(team.TeamID)
@@ -167,7 +169,6 @@ func (cs *EnvironmentServer) RunTurnDefault(team *common.Team) {
 					cs.RemoveAgentFromTeam(agentToAudit)
 				}
 			}
-			cs.ApplyPunishment(team, agentToAudit)
 		}
 
 		for _, agentID := range team.Agents {

